### ISSUE TITLE
Friendly request to `@deprecate` API changes via semver

### ISSUE THREAD

**Issue by dehann** [BioJulia/BioAlignments.jl#84]
Hi BioAlignments,

Just wanted to put in a friendly request for API changes to [follow Julia's `@deprecate`](https://docs.julialang.org/en/v1/base/base/#Base.@deprecate) pattern for API changes please, so for example in package v0.1.x:
```julia
# src/AwesomeCode.jl
oldsignature(x::Int) = # does thing
```

Then the API change goes into v0.2.0 along with the deprecation
```julia
# src/AwesomeCode.jl
newsignature(x::Float64) = # does similar thing

# src/Deprecated.jl
@deprecate oldsignature(x::Int) newsignature(float(x))
```

This will give users a chance to update their scripts with a useful auto-generated warning, which tells them how to replace old calls with new ones.

Then remove the deprecation in v0.3
```julia
# src/AwesomeCode.jl
newsignature(x::Float64) = # does similar thing
```

Also consider listing breaking changes in a `NEWS.md`.  Based on [a quick search](https://github.com/BioJulia/BioAlignments.jl/search?q=%40deprecate) it looks like the `@deprecate` feature is not used here much.

Thanks for the great package!

**Comment by dehann** [comment]
Oh, there is also overwriting the `Base.getproperty(obj::MyType, f::Symbol)` when deprecating fields in a struct.  And then inside that use `Base.getfield` to avoid stack overflows.

Can close this issue on comment received :-)

**Comment by kescobo** [comment]
Yeah, we definitely *should* do this throughout BioJulia, especially after packages have reached v1.0. Apologies for the failure to do so in the past. In principle, we could even do patch-releases on previous versions retroactively, though I'm not sure how much effort that's worth.

Unfortunately, developer time in this ecosystem is pretty limited - I suspect this is why it wasn't done in the past. As an aside, if you want to make a PR for this purpose, I think it would be very warmly received :-)
